### PR TITLE
bootstrap: python3 packages for fedora and redhat based systems

### PR DESCRIPTION
**Pull request by rakeshgm**
adding python3 packages for fedora and RedHat based systems. 

Signed-off-by: rakeshgm <rakeshgm@redhat.com>

### CODE REVIEW

File: bootstrap [hunk 1]
```diff
@@ -45,12 +45,17 @@ Linux)
         fi
         ;;
     RedHatEnterpriseWorkstation|RedHatEnterpriseServer|CentOS)
-        for package in python2-pip python-virtualenv mariadb-devel libev-devel libvirt-devel libffi-devel; do
-	    if [ "$(rpm -q $package)" == "package $package is not installed" ]; then
-		missing="${missing:+$missing }$package"
-	    fi
+        if [[ "$PYTHON" =~ "python2" ]]; then
+            packages=(python-pip python-devel python-virtualenv mariadb-devel libev-devel libvirt-devel libffi-devel)
```
Comment (tchaikov) [inline bootstrap:49]: what's the advantage of using array here?

File: bootstrap [hunk 2]
```diff
@@ -45,12 +45,17 @@ Linux)
...
+        else
+            pacakges=(python3-pip python3-devel python3-virtualenv mariadb-devel libev-devel libvirt-devel libffi-devel)
+        fi
+        for package in  ${packages[@]}; do
```
Comment (tchaikov) [inline bootstrap:53]: please remove the extra space.

Comment (rakeshgm) [reply bootstrap]: done

File: bootstrap (hunk 1 above)
Comment (rakeshgm) [reply bootstrap:49]: I don't this is an array here. I think this is a conditional construct. [ref here](http://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Conditional-Constructs).

Comment (tchaikov) [reply bootstrap:49]: i don't think i am talking about `if`. what i am referencing is the above line.
```sh
packages=(....)
```

> Arrays are assigned to using compound assignments of the form
> name=(value1 value2 … )

see https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Arrays

Comment (rakeshgm) [reply bootstrap:49]: ok. correct me If I am wrong. the pattern I followed is similar to that one where the comparison for previous distros already exists. [some thing like this](https://github.com/ceph/teuthology/blob/master/bootstrap#L24)

Comment (kshtsk) [reply bootstrap:49]: > what's the advantage of using array here?

Why not using array for lists? It is natural and allows to generalize iteration algorithms:
```
a=("abc" "a b c")

for i in "${a[@]}" ; do
  echo - $i
done

b='xyz "x y z"'
for i in $b ; do
  echo + $i
done
```
will give:
```
- abc
- a b c
+ xyz
+ "x
+ y
+ z"
```

Also, each element of array can be taken by index.

**Comment by kshtsk** [comment]
@rakeshgm Overall change looks fine to me, I believe you tested this on fedora an rhel?

File: bootstrap (hunk 1 above)
Comment (kshtsk) [reply bootstrap:49]: > ok. correct me If I am wrong. the pattern I followed is similar to that one where the comparison for previous distros already exists. [some thing like this](https://github.com/ceph/teuthology/blob/master/bootstrap#L24)

Maybe use the same pattern as you refer to, like use `$deps` instead `$packages`?

Comment (rakeshgm) [reply bootstrap]: done. changed `deps`

**Comment by rakeshgm** [comment]
> @rakeshgm Overall change looks fine to me, I believe you tested this on fedora an rhel?

yes.

**Comment by kshtsk** [comment]
@susebot run deploy

**Comment by kshtsk** [comment]
The failure is not related to the change, though, scheduling another run...

**Comment by kshtsk** [comment]
@susebot run deploy

File: bootstrap (hunk 1 above)
Comment (vasukulkarni) [reply bootstrap]: Resolving, I think rakesh is following same patter what we have in the rest of the places in bootstrap.

**Pull request merged**